metricbeat/module/system/cpu/metrics/metrics_freebsd.go

```diff
@@ -0,0 +1,66 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package metrics
+
+import (
+	"bufio"
+	"strings"
+
+	"github.com/joeshaw/multierror"
+	"github.com/pkg/errors"
+)
+
+func scanStatFile(scanner *bufio.Scanner) (CPUMetrics, error) {
+	cpuData, err := statScanner(scanner, parseCPULine)
+	if err != nil {
+		return CPUMetrics{}, errors.Wrap(err, "error scanning stat file")
+	}
+	return cpuData, nil
+}
+
+func parseCPULine(line string) (CPU, error) {
+	cpuData := CPU{}
+	fields := strings.Fields(line)
+	var errs multierror.Errors
+
+	user, err := touint(fields[1])
+	if err != nil {
+		errs = append(errs, err)
+	}
+	cpuData.user = &user
```

Comment:
Can we trust the value returned by `touint` in case of error? Should we set the value only if there was no error?
```suggestion
	user, err := touint(fields[1])
	if err != nil {
		errs = append(errs, err)
	} else {
		cpuData.user = &user
	}
```

Reply:
> Can we trust the value returned by touint in case of error? Should we set the value only if there was no error?

This is more a quirk of the library. It doesn't like it when you append `nil` errors.

Reply:
Yep, I am ok with adding `err` to the list of errors only if it is not nil. But do we want to use the returned int if there was an error?

Reply:
> Yep, I am ok with adding err to the list of errors only if it is not nil. But do we want to use the returned int if there was an error?

Theoretically if there's an error converting that value to an int, the values in `/proc/stat` aren't what we expect, and we should consider the data bad.

Reply:
Oh, ok, so it is up to the caller to decide what to do with the cpu data if some error was returned :+1: